samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -123,17 +123,60 @@ public synchronized String getEphemeralPath() {
   }
 
   /**
-   * Method is used to get the <i>sorted</i> list of currently active/registered processors
+   * Method is used to get the <i>sorted</i> list of currently active/registered processors (znodes)
    *
    * @return List of absolute ZK node paths
    */
   public List<String> getSortedActiveProcessors() {
-    List<String> children = zkClient.getChildren(keyBuilder.getProcessorsPath());
-    if (children.size() > 0) {
-      Collections.sort(children);
-      LOG.info("Found these children - " + children);
+    List<String> znodeIds = zkClient.getChildren(keyBuilder.getProcessorsPath());
+    if (znodeIds.size() > 0) {
+      Collections.sort(znodeIds);
+      LOG.info("Found these children - " + znodeIds);
     }
-    return children;
+    return znodeIds;
+  }
+
+  /**
+   * Method is used to read processor's data from the znode
+   * @param fullPath absolute path to the znode
+   * @return processor's data
+   */
+  public String getProcessorsData(String fullPath) {
+    String data = zkClient.<String>readData(fullPath, true);
+    if (data == null) {
+      throw new SamzaException(String.format("Cannot read ZK node:", fullPath));
+    }
+    return data;
+  }
+
+  /**
+   * Method is used to get the <i>sorted</i> list of currently active/registered processor ids
+   * @return List of processorIds
+   */
+  public List<String> getSortedActiveProcessorsIDs() {
+    return getSortedActiveProcessorsIDs(getSortedActiveProcessors());
+  }
+
+  /**
+   * Method is used to get the <i>sorted</i> list of processors ids for a given list of znodes
+   * @param znodeIds - list of relative paths of the children's znodes
+   * @return List of processor ids for a given list of znodes
+   */
+  public List<String> getSortedActiveProcessorsIDs(List<String> znodeIds) {
+    String processorPath = keyBuilder.getProcessorsPath();
+    List<String> processorIds = new ArrayList<>(znodeIds.size());
+    if (znodeIds.size() > 0) {
+
+      for (String child : znodeIds) {
+        String fullPath = String.format("%s/%s", processorPath, child);
+        processorIds.add(getProcessorsData(fullPath));
+      }
+
+      Collections.sort(processorIds);
```

Comment:
I don't think you should sort this list again. The input is a list of sorted znodeIds. "processorIds" is a list of data stored in the znode. Sorting here will re-order everything. 

Please change the name of the method appropriately.